spec/activerecord-bitemporal/bitemporal_spec.rb

```diff
@@ -1773,4 +1773,119 @@ class EmployeeWithUniquness < Employee
       end
     end
   end
+
+  describe "#bitemporal_build_update_records" do
+    let(:from) { time_current }
+    let(:to) { from + 10.days }
+    let(:finish) { Time.utc(9999, 12, 31).in_time_zone }
+    let!(:employee) { Employee.create!(name: "Jone", valid_from: from, valid_to: to) }
+    let(:now) { time_current }
+    let(:valid_datetime) { time_current }
+    let(:force_update) { false }
+
+    subject do
+      employee.name = "Tom"
+      employee.__send__(:bitemporal_build_update_records, valid_datetime: valid_datetime, current_time: now, force_update: force_update)
+    end
+
+    # before: |-----------------------|
+    # update:             *
+    # after:  |-----------|
+    #                     |-----------|
+    context "valid_datetime is in" do
+      let(:valid_datetime) { from + 5.days }
+
+      it "returns the current valid record and before/after records" do
+        current, before, after = subject
+
+        aggregate_failures do
+          expect(current).to have_attributes(transaction_to: now, deleted_at: now)
+          expect(before).to have_attributes(name: "Jone", valid_from: from, valid_to: valid_datetime, transaction_from: now)
+          expect(after).to have_attributes(name: "Tom", valid_from: valid_datetime, valid_to: to, transaction_from: now)
```

Comment:
[nits] It would be better to test all of `name` `valid_from` `valid_to` `transaction_from` `transaction_to` (and `deleted_at` too ?).
e.g.
```ruby
expect(current).to have_attributes(name: "Jone",  valid_from: from, valid_to: to, transaction_from: created_by_employee, transaction_to: now, deleted_at: now)
expect(before).to have_attributes(name: "Jone", valid_from: from, valid_to: valid_datetime, transaction_from: now, transaction_to: finish)
expect(after).to have_attributes(name: "Tom", valid_from: valid_datetime, valid_to: to, transaction_from: now, transaction_to: finish)
```

Reply:
Fixed https://github.com/kufu/activerecord-bitemporal/pull/75/commits/62252612d922a82d9a88e7124055458a14e90673